airbyte-integrations/bases/base-normalization/dbt-project-template/macros/configuration.sql

```diff
@@ -6,7 +6,12 @@
     {%- set schemaname, _, tablename = var("models_to_source")[this.identifier].partition(".") -%}
 
     {%- call statement("get_column_type", fetch_result=True) -%}
-        set search_path to '$user', public, {{ schemaname }};
+        {%- set public_schema_query -%}
+            SELECT COUNT(1) AS count FROM pg_namespace WHERE nspname = 'public'
+        {%- endset -%}
+        {%- set public_schema_count = run_query(public_schema_query).rows[0]['count'] -%}
+
+        set search_path to '$user', {%- if public_schema_count > 0 -%} public, {%- endif -%} {{ schemaname }};
```

Comment:
I think we can skip all this detection and just remove "public" schema from this search_path

Reply:
is this because if the user wants to use the `public` schema, they would explicitly pass that to us? I.e. `schemaname` would be set to `public`?

Reply:
in really I think I was wrong, we need it 

I am playing now with better solution which store old `search_path`, query pg_table_def and restore old `search_path` back